model/versioning/Versioned.php

```diff
@@ -301,12 +308,13 @@ public function augmentSQL(SQLSelect $query, DataQuery $dataQuery = null) {
 		// Reading a specific stage (Stage or Live)
 		case 'stage':
 			$stage = $dataQuery->getQueryParam('Versioned.stage');
-			if($stage && ($stage != $this->defaultStage)) {
+			if($this->hasVersionedStage($stage) && $stage !== static::DRAFT_STAGE) {
```

Comment:
if it doesn't have stage `$stage` then we silently fail? shouldn't we throw something or at least warn?